junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/StringToObjectConverter.java

```diff
@@ -14,39 +14,33 @@
  * Internal API for converting arguments of type {@link String} to a specified
  * target type.
  */
-interface StringToObjectConverter {
+abstract class StringToObjectConverter implements Converter {
+
+	@Override
+	public final boolean canConvert(Object source, TypeDescriptor targetType) {
+		return canConvert(targetType.getType());
+	}
 
 	/**
 	 * Determine if this converter can convert from a {@link String} to the
 	 * supplied target type (which is guaranteed to be a wrapper type for
 	 * primitives &mdash; for example, {@link Integer} instead of {@code int}).
 	 */
-	boolean canConvertTo(Class<?> targetType);
+	abstract boolean canConvert(Class<?> targetType);
 
-	/**
-	 * Convert the supplied {@link String} to the supplied target type (which is
-	 * guaranteed to be a wrapper type for primitives &mdash; for example,
-	 * {@link Integer} instead of {@code int}).
-	 *
-	 * <p>This method will only be invoked in {@link #canConvertTo(Class)}
-	 * returned {@code true} for the same target type.
-	 */
-	Object convert(String source, Class<?> targetType) throws Exception;
```

Comment:
Being an internal API, I deleted this `convert(String, Class<?>)` variant to reduce complexity. The downside is that the `ClassLoader` parameter is unused on average.